### PR TITLE
chore(docs): Conditionally sourcing files using environment variables

### PR DESCRIPTION
**Pull request by baranbbr**
## Description

Adds a section to `sourcing-from-filesystem` on how environment variables can be used to source files.
For example, you might do this if you want to source a smaller batch of files when in development, to save on local build times.

If/when these changes get approved, It might be worth also mentioning them here: [gatsbyjs.com/docs/how-to/performance/improving-build-performance/](https://www.gatsbyjs.com/docs/how-to/performance/improving-build-performance/).

Reference for changes:
- [paulie.dev/posts/2021/07/gatsby-slow-local-build-times/](https://paulie.dev/posts/2021/07/gatsby-slow-local-build-times/) @PaulieScanlon

### CODE REVIEW

File: docs/docs/how-to/sourcing-data/sourcing-from-the-filesystem.md [hunk 1]
```diff
@@ -96,6 +96,31 @@ You can then update your query using `sourceInstanceName` and the value of the `
 }
 ```
 
+## Sourcing Files using Environment Variables
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-the-filesystem.md:99]:
```suggestion
## Conditionally sourcing files using environment variables
```

One is not really "using" environment variables but is using them to conditionally set `path`

File: docs/docs/how-to/sourcing-data/sourcing-from-the-filesystem.md [hunk 2]
```diff
@@ -96,6 +96,31 @@ You can then update your query using `sourceInstanceName` and the value of the `
...
+
+You can also source files using environment variables. For context, you might decide to do this if you're sourcing a lot of files and local build times are taking too long, so you only source a smaller batch of files when in development.
+
+In the example below in `gatsby-config.js` if the environment variable `NODE_ENV` is set to development, only the content in the specified path will be sourced.
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-the-filesystem.md:103]:
```suggestion
You can conditionally set the `path` option using environment variables. For context, you might decide to do this if you're sourcing a lot of files and you're interested in only sourcing a smaller batch of files during `gatsby develop`. This is also helpful when you e.g. have a staging and production environment (signaled through environment variables).

The example below shows how to use `NODE_ENV` (which is automatically set to `development` during `gatsby develop`) to only source a smaller portion of the content during development. For `gatsby build` the full dataset will be used.
```

Wanted to point out that the key here is to conditionally set the `path` option.

File: docs/docs/how-to/sourcing-data/sourcing-from-the-filesystem.md [hunk 3]
```diff
@@ -96,6 +96,31 @@ You can then update your query using `sourceInstanceName` and the value of the `
...
+
+```javascript:title=gatsby-config.js
+require('dotenv').config({
+  path: `.env.${process.env.NODE_ENV}`,
+})
+
+module.exports = {
```
Comment (LekoArts) [inline docs/docs/how-to/sourcing-data/sourcing-from-the-filesystem.md:110]:
```suggestion
module.exports = {
```

The `dotenv` is not necessary as `NODE_ENV` will be set by Gatsby automatically

**Review by LekoArts — CHANGES_REQUESTED** [review]
Thanks for the PR! I've got a couple of improvements :)

**Comment by baranbbr** [comment]
Amazing, thanks for the changes @LekoArts ❤️

Before merging, is it worth mentioning this in the improving build performance section linked above?

**Comment by baranbbr** [comment]
Done! Hope that's okay and the link to [gatsbyjs.com/docs/how-to/sourcing-from-the-filesystem](https://www.gatsbyjs.com/docs/how-to/sourcing-data/sourcing-from-the-filesystem/) works 😅

**Review by LekoArts — APPROVED** [review]
Thanks!

**Pull request merged**